src/main/webapp/app/guided-tour/guided-tour.service.ts

```diff
@@ -877,14 +877,12 @@ export class GuidedTourService {
             return false;
         }
         const currentTourStep = this.currentTour.steps[this.currentTourStepIndex];
-        const selector = currentTourStep.highlightSelector;
-        if (selector) {
-            const selectedElement = document.querySelector(selector);
-            if (!selectedElement) {
-                return false;
-            }
+        if (!currentTourStep) {
+            this.resetTour();
+            return false;
         }
-        return true;
+
+        return !(!currentTourStep.highlightSelector || !document.querySelector(currentTourStep.highlightSelector));
```

Comment:
Is this intentionally written like this? Otherwise I'd use de-morgan to get rid of all the `!` to reduce head-arithmetics:
```suggestion
        return currentTourStep.highlightSelector && document.querySelector(currentTourStep.highlightSelector);
```